src/parser-babylon.js

```diff
@@ -25,12 +26,14 @@ function parse(text) {
     ]
   };
 
+  const parseMethod = options.parseExpression ? "parseExpression" : "parse";
```

Comment:
Yeah, I backed out the `parseJavaScriptExpression` change, though, since the JSON parser doesn't use `babylon.parseExpression` anymore.